matRad_DijSampling.m

```diff
@@ -35,25 +34,63 @@
 %
 % %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
-if ~exist('SamplingRate','var')
-    SamplingRate       = 0.1;
+boolFineClustering  = false;                                     % flag to switch between fine and coarse radiological depth resolution
+deltaRadDepth       = 5;                                         % step size of radiological depth
+LatCutOffsq         = r0^2;
+
+%% remember dose values inside the inner core
+ixCore              = rad_distancesSq < LatCutOffsq;             % get voxels indices having a smaller radial distance than r0
+bixelDoseCore       = bixelDose(ixCore);                         % save dose values that are not affected by sampling
+
+ixTail              = ~ixCore;                                   % get voxels indices beyond r0
+linIxSample         = find(ixTail);                              % convert logical index to linear index
+numTail             = numel(linIxSample);
+bixelDoseTail       = bixelDose(linIxSample);                    % dose values that are going to be reduced by sampling
+ixSampTail          = ix(linIxSample);                           % indices that are going to be reduced by sampling
+
+%% sample for each radiological depth the lateral halo dose  
+radDepthTail        = (radDepthV(linIxSample));                  % get radiological depth in the tail
+
+% cluster radiological dephts to reduce computations
+B_r                 = int32(ceil(radDepthTail));                 % cluster radiological depths; hist(B,NumOfClusters) could also be used
```

Comment:
If you get rid of boolFineClustering you don't need the rounding to integers anymore